lib/cpp/src/thrift/concurrency/PosixThreadFactory.h

```diff
@@ -39,7 +39,14 @@ class PosixThreadFactory : public ThreadFactory {
   /**
    * POSIX Thread scheduler policies
    */
-  enum POLICY { OTHER, FIFO, ROUND_ROBIN };
+  class Policy {
+  public:
+    enum value {
+      OTHER       = 0,
+      FIFO        = 1,
+      ROUND_ROBIN = 2
+    };
```

Comment:
I understand your point, although I wouldn't personally do this as it feels fighting against language.
If we do this, IMO we should introduce implicit conversion from inner enum to outer class to remove the `::value` in method signatures.
But I think it's completely separate issue and should be handled separately from this patch.